code/modules/mob/living/simple_animal/hostile/space_dragon.dm

```diff
@@ -152,6 +148,16 @@
 	if(istype(target, /obj/vehicle/sealed/mecha))
 		var/obj/vehicle/sealed/mecha/M = target
 		M.take_damage(50, BRUTE, MELEE, 1)
+	
+/mob/living/simple_animal/hostile/space_dragon/RangedAttack(atom/target, modifiers)
```

Comment:
Same advice on the spider PR. I think this should be abstracted into a `ranged_attack_secondary`.

I would recommend making that its own PR, which I can merge before these two as soon as it's ready.